astropy/io/fits/file.py

```diff
@@ -400,6 +404,36 @@ def _overwrite_existing(self, overwrite, fileobj, closed):
             else:
                 raise IOError("File {!r} already exists.".format(self.name))
 
+    def _test_for_compression(self, obj_or_name, magic, mode, ext=None):
```

Comment:
I was thinking about ordering operators in general (which throw exceptions with incompatible types) and forgot that `==` and `!=` do work even if the types aren't orderable.